apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -176,6 +176,28 @@ type ResourceQuota struct {
 // ResourceFlavorReference is the name of the ResourceFlavor.
 type ResourceFlavorReference string
 
+// ClusterQueuePendingWorkload contains the information identifying a pending workload
+// in the cluster queue.
+type ClusterQueuePendingWorkload struct {
+	// Name indicates the name of the pending workload.
+	Name string `json:"name"`
+
+	// Namespace indicates the name of the pending workload.
+	Namespace string `json:"namespace"`
+}
+
+type ClusterQueuePendingWorkloadsStatus struct {
+	// Head contains the list of top pending workloads.
+	// +listType=map
+	// +listMapKey=name
+	// +listMapKey=namespace
```

Comment:
```suggestion
	// +listType=atomic
```

The order here matters, so it should not be a map.

Reply:
changed to +listType=atomic